poc_iot_verifier/src/loader.rs

```diff
@@ -150,53 +152,46 @@ impl Loader {
     async fn handle_store_update(&self, file_type: FileType, buf: &[u8]) -> Result {
         match file_type {
             FileType::LoraBeaconIngestReport => {
-                let beacon = LoraBeaconIngestReportV1::decode(buf)?;
-                let event = beacon.report.unwrap();
-                let packet_data = event.data.clone();
-                let buf_as_vec = buf.to_vec();
-                let mut public_key = event.pub_key.clone();
-                // TODO: maybe this ID construction can be pushed out to a trait or part of the report struct ?
-                let mut id: Vec<u8> = packet_data.clone();
-                id.append(&mut public_key);
-                let id_hash = Sha256::digest(&id).to_vec();
+                let beacon: LoraBeaconIngestReport =
+                    LoraBeaconIngestReportV1::decode(buf)?.try_into()?;
+                tracing::debug!("beacon report from ingestor: {:?}", &beacon);
+                let packet_data = beacon.report.data.clone();
+                tracing::debug!("beacon data: {:?}", &packet_data);
                 Report::insert_into(
                     &self.pool,
-                    id_hash,
+                    beacon.generate_id(),
                     packet_data,
-                    buf_as_vec,
-                    &event.timestamp()?,
```

Comment:
why are we undoing the MsgTimestamp trait here?

Reply:
changed to using the received_timestamp  from the beacon report, this is already a DateTime as a result of the LoraBeaconIngestReportV1 -> LoraBeaconIngestReport at line 155.